src/notification/NotificationController.ts

```diff
@@ -0,0 +1,266 @@
+import type { Patch } from 'immer';
+import { nanoid } from 'nanoid';
+
+import { hasProperty } from '../util';
+import { BaseController } from '../BaseControllerV2';
+
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+
+/**
+ * @type NotificationState
+ * @property notifications - Stores existing notifications to be shown in the UI
+ */
+export type NotificationState = {
+  notifications: Record<string, Notification>;
+};
+
+export type Notification = {
+  id: string;
+  type: NotificationType;
+  origin: string;
+  date: number;
+  read: boolean;
```

Comment:
Would it be useful to track when a notification was seen? If we change this from a boolean to a number representing the time the notification was read, this field could serve both purposes.